File: algorithms/data-structures/arrays/arrays-ds.js
// Arrays - DS
// https://www.hackerrank.com/challenges/arrays-ds/problem

function reverseArray(a) {
  /*
  let reversed = '';
  for (let char of a) {
    reversed = char + ' ' + reversed;
  }
  return reversed;
  */
  // return a.reduce((prev, curr) => curr + ' ' + prev, '');
  return a.reverse().join(' ');
}

File: algorithms/data-structures/arrays/Arrays - DS/arrays-ds.py
# Arrays - DS
# https://www.hackerrank.com/challenges/arrays-ds/problem

# def reverseArray(a):
#   return reversed(a)

# def reverseArray(a):
#   return a[::-1]

def reverseArray(a):
  result = []
    
  for n in range(len(a) - 1, -1, -1):
    result.append(a[n])

  return result